lib/constructs/cost-api-implementation.ts

```diff
@@ -0,0 +1,73 @@
+import { Construct } from "constructs";
+import * as lambdaNodeJs from "aws-cdk-lib/aws-lambda-nodejs";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { Duration } from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { AtatQueue } from "./sqs";
+import { AtatRestApi } from "./apigateway";
+import { HttpMethod } from "../http";
+import * as apigw from "aws-cdk-lib/aws-apigateway";
+import { IApiRoute } from "./api-route";
+import { IResource, Method } from "aws-cdk-lib/aws-apigateway";
+
+export interface CostResourcesProps {
+  readonly environmentName: string;
```

Comment:
At some point (not a thing for this PR), I want to see if we need to actually pass this around all over the place all the time...